iradix.go

```diff
@@ -47,12 +60,16 @@ func (t *Tree) Txn() *Txn {
 func (t *Txn) writeNode(n *Node) *Node {
 	// Ensure the modified set exists
 	if t.modified == nil {
-		t.modified = make(map[*Node]struct{})
+		lru, err := simplelru.NewLRU(defaultModifiedCache, nil)
```

Comment:
Looking through here I think it makes sense that this sometimes will return a fresh copy of something that's already been modified, since the caller can never know if it came from the cache or not, they can't care.